### PR TITLE
Update function_space build to include surface topography modification

### PR DESCRIPTION
**Pull request by akshaysridhar**
Summary : Updates function space construction to allow inclusion of topography. 
Initial commits that replicate test cases for topo construction in ClimaCore.
PRs with additional examples and demonstrations in "running" cases will
follow. 
Changes:
- [x] Update make_function_space flags to allow cases with warped surfaces to be built
- [x] Include containers for "canonical" or "warped" surfaces
- [x] Update show functions for topography characteristics
- [x] Link to ClimaCore Hypsography packages for future extensions.  
- [x] Track 0.9.0 version ; Update Project.toml 
- [ ] Add demo plots + Documentation

Status: 
Pending bug-fixes in `ClimaCore.jl`

### CODE REVIEW

**Comment by akshaysridhar** [comment]
bors try

File: src/Domains/topography.jl [hunk 1]
```diff
@@ -0,0 +1,29 @@
+abstract type AbstractTopography end
+
+"""
+  CanonicalSurface <: AbstractTopography
```
Comment (bischtob) [inline src/Domains/topography.jl:4]: Is there a reason why we have both CanonicalSurface and NoTopgraphy?

Comment (akshaysridhar) [reply src/Domains/topography.jl:4]: Nope - changing names to only support `WarpedSurface` and `CanonicalSurface`

**Comment by bischtob** [comment]
@akshaysridhar , what is the status here?

**Comment by akshaysridhar** [comment]
Superseded by #535

**Pull request closed without merging**